csharp/ql/src/utils/model-generator/internal/CaptureModelsSpecific.qll

```diff
@@ -69,10 +69,16 @@ predicate isRelevantType(CS::Type t) {
  */
 string qualifierString() { result = "Argument[Qualifier]" }
 
+private string getTypeBasedSuffix(Type t) {
+  if Collections::isCollectionType(t) then result = ".Element" else result = ""
+}
+
 private string parameterAccess(CS::Parameter p) {
-  if Collections::isCollectionType(p.getType())
-  then result = "Argument[" + p.getPosition() + "].Element"
-  else result = "Argument[" + p.getPosition() + "]"
+  result = "Argument[" + p.getPosition() + "]" + getTypeBasedSuffix(p.getType())
+}
+
+private string returnValueAccess(CS::Callable c) {
+  result = "ReturnValue" + getTypeBasedSuffix(c.getReturnType())
```

Comment:
This only makes sense of the generated row is a `value` step and not a `taint` one.